napari/layers/labels/labels.py

```diff
@@ -766,7 +766,7 @@ def paint(self, coord, new_label, refresh=True):
             self._save_history()
         brush_size_dims = [self.brush_size] * self.ndim
         # Scale the brush size from world coorinates to data coordinates
-        brush_size_dims = np.divide(brush_size_dims, self.scale)
+        # brush_size_dims = np.divide(brush_size_dims, self.scale)
```

Comment:
Can you explain why this was needed?

Reply:
Ah looking at the change above, it looks like something was going wrong when the labels layer was no longer active or something? Can you produce a minimal napari example with a set of reproducible steps and a screenshot to demonstrate why this was necessary. I can sort of imagine what caused it, but would be good to see explicitly and then maybe be able to add a test

Reply:
because brush size is not scaled. So If you set brush size to 10 and have image with scale (200, 70, 70) it cannot paint anything. Other option is to update range of brush size range.

Reply:
is that because of the range on the slider? i'll have to take a look later, but that's helpful to know

Reply:
The maximum size of the brush is 40. I work with data with spacing (210, 70, 70) so never have brush size above 1. 

To have a similar experience independently of scaling factor I assume that brush size should be in pixels (to have simper usage from API side). 

If the brush size should be in some units there should be a different default value base on current scaling, Also slider range should be updated. Everything also should be updated on layer scale update.

Reply:
ok yeah, that makes sense, that 40 definitely doesn't make sense in this context

Reply:
I did like having the brush size in "world coordinates", so after the scaling. Could we take another approach were the slider stays in pixels but brush size itself is transformed? I.e. put the scaling in between the brush size and the slider? I think that would fix things too and wouldn't require changing any of our tests or labels code

Reply:
As brush size is set independently for each label layer could you point the benefits of using them? 
For me, more natural is to use a pixel-based scale (for example when set `layer.brush_size = 20` not `layer.brush_size = 20 * 70.74`(nm) or `layer.brush_size = 20 * 70.74 * 10**-9`(m). 
As I look at the image I count visible pixels. 

But maybe there are some benefits missed by me.

Reply:
> As I look at the image I count visible pixels.

Alright, that's a fair argument. You count pixels in the data space. I'll take a closer look, but I think it's fairly close then. I might do a little clean up, like remove the commented line, and we might be able to reduce some code duplication in the tests now etc.

Curious if @jni wants to share any perspective on this too before finishing?

Reply:
I see on point which may need to be addressed is painting with different spacing in dimensions.

For example, I work on data with 210, 70, 70 nm spacing. Lets define brush_size on 20. The important question is if it should span along z-axis on 20 pixels or only 7. The same problems may show when the user sets different spacing for the visible axis. 

I meet both scenarios. Approach which I use is to implement normalization by the shortest visible axis

Future note: When introducing units then also need to handle the time axis.

Reply:
I don't have strong opinions about this... I can see arguments for brush size to be both in world and pixel coordinates, but pixel coordinates are less likely to blow up on our faces I think. The only counter-argument I have is support for anisotropic brushes is nice. I can't follow the code clearly enough to know whether this gives me an ellipse in anisotropic visible space...

Reply:
Ok thanks for weighing in @jni - let's stick with data/ pixel coordinates for the brush. I've adjusted the implementation slightly so that while the brush is now in data coordinates again, the cursor stays in world coordinates.